Sources/Tracking/CustomerIOImplementation.swift

```diff
@@ -70,28 +70,35 @@ public class CustomerIOImplementation: CustomerIOInstance {
         body: RequestBody,
         jsonEncoder: JSONEncoder? = nil
     ) {
-        let isChangingIdentifiedProfile = profileStore.identifier != nil && profileStore.identifier == identifier
-        if isChangingIdentifiedProfile {
-            // TODO: add to background queue delete device token from currently registered profile.
+        let currentlyIdentifiedProfileIdentifier = profileStore.identifier
+        let isChangingIdentifiedProfile = currentlyIdentifiedProfileIdentifier != nil &&
+            currentlyIdentifiedProfileIdentifier != identifier
+
+        if let currentlyIdentifiedProfileIdentifier = currentlyIdentifiedProfileIdentifier,
+           isChangingIdentifiedProfile {
+            hooks.profileIdentifyHooks.forEach { hook in
+                hook.beforeNewProfileIdentified(oldIdentifier: currentlyIdentifiedProfileIdentifier,
```

Comment:
not sure the name is super expressive. `beforeIdentifiedProfileChange` maybe?

Reply:
Yeah, I like that. As long as the name is suggesting that we are *changing to a new profile* when there was a previous profile identified, that will work.